pageserver/src/tenant/metadata.rs

```diff
@@ -70,6 +71,7 @@ struct TimelineMetadataBodyV3 {
     latest_gc_cutoff_lsn: Lsn,
     initdb_lsn: Lsn,
     pg_version: u32,
+    last_aux_file_policy: Option<AuxFilePolicy>,
```

Comment:
Let's make this more generic: there will be other cases in future where we want to make some change to the format version (e.g. for relation size info), and it's simpler to test if we have version 1,2,3 rather than arbitrary combinations of flags.

I think we end up with several versions in the system:
- Existing: `IndexPart::version` (the version of only the index struct)
- Existing: `Summary::format_version` (the version of an individual layer file)
- New: `storage_version` (or whatever we call it), which describes high level changes to the storage format such as how we map keys to metadata.

(this is a separate concern to whether we store this in TimelineMetadata or in IndexPart -- I think Joonas's suggestion to store it in IndexPart has merit here https://github.com/neondatabase/neon/pull/7663#pullrequestreview-2049523569)

Reply:
Decided not to do that for now.